libs/@local/graph/authorization/src/policies/components.rs

```diff
@@ -51,6 +189,7 @@ pub struct PolicyComponentsBuilder<'a, S> {
     entity_type_ids: HashSet<Cow<'a, VersionedUrl>>,
     entity_edition_ids: HashSet<EntityEditionId>,
     actions: HashSet<ActionName>,
+    enable_optimization: bool,
```

Comment:
Does it make sense to even gate this? What would be scenarios where we'd want to disable the optimizations?